src/storage/src/hummock/utils.rs

```diff
@@ -606,10 +620,17 @@ pub(crate) async fn wait_for_epoch(
                 return Err(HummockError::wait_epoch("tx dropped").into());
             }
             Ok(Ok(_)) => {
-                let max_committed_epoch = *receiver.borrow();
-                if max_committed_epoch >= wait_epoch {
+                // TODO: should handle the corner case of drop table
```

Comment:
If we don't handle the corner case in this PR, does it mean that `wait_for_epoch` will be blocked forever? For example, when a batch query is issued to CN but before execution the table is dropped, will the query be blocked at `waif_for_epoch` forever?

Reply:
Just updated the code. 

A separate `BatchQueryCommitted` variant is introduced to `HummockReadEpoch` to provide an extra hummock version id to wait on for batch query. In the hummock version of the provided id, the table id must have been registered to hummock. In this way,  if the table id does not exist in a later version, the table must have been dropped, and then we will return with an error.

For the wait_epoch in streaming executor, we can assume that when the streaming executor exists, its streaming job will prevent the table from being dropped, and therefore when we see that a table id does exist in the latest version, the table must not be already dropped, and then we can wait for the table id be registered in later version.